exasol/ai/mcp/server/mcp_server.py

```diff
@@ -377,17 +385,46 @@ def execute_query(
         raise ValueError("The query is invalid or not a SELECT statement.")
 
 
-if __name__ == "__main__":
+def get_mcp_settings() -> McpServerSettings:
+    """
+    Reads optional settings. They can be provided either in a json string stored in the
+    EXA_MCP_SETTINGS envar or in a json file. In the latter case EXA_MCP_SETTINGS must
+    contain the file path.
+    """
+    try:
+        settings_text = os.environ.get(ENVAR_SETTINGS)
+        if not settings_text:
+            return McpServerSettings()
+        elif re.match(r"^\s*\{.*\}\s*$", settings_text):
+            return McpServerSettings.model_validate_json(settings_text)
+        elif os.path.isfile(settings_text):
```

Comment:
I think, the regex is not needed, you could simply call `McpServerSettings.model_validate_json(settings_text)` and catch the exception. and then check in the except block for the path.